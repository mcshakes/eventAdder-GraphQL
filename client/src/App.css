.main-content {
  margin: 4rem 2.5rem; 
  /*so it's right below the nav*/
  
}