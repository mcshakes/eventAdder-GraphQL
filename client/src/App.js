import React from 'react';
import { BrowserRouter, Route, Redirect, Switch } from "react-router-dom";

import AuthPage from "./components/Auth";
import BookingsPage from "./components/Bookings";
import EventsPage from "./components/Events";

import MainNavigation from "./components/navigation/MainNavigation";
import AuthContext from "./context/auth-context";

import './App.css';

class App extends React.Component {
  
  login = (token, userId, tokenExpiration) => {

  }


  render() {
    return (
      <BrowserRouter>

        <AuthContext.Provider value={{token: null, userId: null}}>
          <MainNavigation />

          <main className="main-content">
            <Switch>
              <Redirect from="/" to="/auth" exact/>
              <Route path="/auth" component={ AuthPage } />
              <Route path="/events" component={ EventsPage } />
              <Route path="/bookings" component={ BookingsPage } />
            </Switch>
            
          </main>
        </AuthContext.Provider>

      </BrowserRouter>

    )
  };
}

export default App;
