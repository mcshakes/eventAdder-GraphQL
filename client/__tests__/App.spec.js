import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import App from '../src/App';

afterEach(cleanup)

describe('App', () => {
    test('renders App component', () => {
      const { asFragment } = render(<App />)
    
      expect(asFragment(<App />)).toMatchSnapshot()
    });
});

// describe('true is truthy and false is falsy', () => {
//     test('true is truthy', () => {
//       expect(true).toBe(true);
//     });
   
//     test('false is falsy', () => {
//       expect(false).toBe(false);
//     });
//   });