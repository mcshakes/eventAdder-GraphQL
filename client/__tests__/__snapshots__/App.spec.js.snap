// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<DocumentFragment>
  <header
    class="main-navigation"
  >
    <div
      class="main-navigation__logo"
    >
      <h1>
        Events! Book Em!
      </h1>
    </div>
    <nav
      class="main-navigation__items"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/auth"
          >
            Log In
          </a>
        </li>
        <li>
          <a
            href="/events"
          >
            Browse Events
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="main-content"
  >
    <form
      class="auth-form"
    >
      <div
        class="form-control"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="form-control"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <div
        class="form-actions"
      >
        <button
          data-testid="auth-submit"
          type="submit"
        >
          Submit
        </button>
        <button
          data-testid="auth-selection"
          type="button"
        >
          Switch to Sign Up
        </button>
      </div>
    </form>
  </main>
</DocumentFragment>
`;
